.travis.yml

```diff
@@ -0,0 +1,19 @@
+# Force OS X.
+language: objective-c
+os: osx
+osx_image: xcode7 # 10.10
+matrix:
+  include:
+    - osx_image: beta-xcode6.2 # 10.9
+
+rvm:
+  - rvm use system
+
+install:
+  - rvm use system
+  - cd /usr/local; git fetch --unshallow
+  - cd /usr/local; [ "$TRAVIS_PULL_REQUEST" != "false" ] && git fetch origin +refs/pull/$TRAVIS_PULL_REQUEST/merge
+  - cd /usr/local; git reset --hard $TRAVIS_COMMIT
```

Comment:
Personally, I think the symlink solution(like below) is better than git manipulation. It's easier, more robust and with better performance.

```
install:
   - rm /usr/local/bin/brew
   - ln -sf $PWD/bin/brew /usr/local/bin/brew
   - mkdir -p /usr/local/Cellar
```

It replaces `/usr/local/bin/brew` to a symlink to the brew file in the test repo. Doing so, Homebrew will use travis's repo as library while preserving  `/usr/local/` as prefix.